backend/src/test/java/de/silvia/backend/controller/ArtikelListControllerTest.java

```diff
@@ -0,0 +1,182 @@
+import de.silvia.backend.security.services.UserService;
+import de.silvia.backend.services.ArtikelListService;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+
+import java.security.Principal;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class ArtikelListControllerTest {
+
+    private final IArtikelListRepo testArtikelListRepo = mock(IArtikelListRepo.class);
+    private final ArtikelListService mockedArtikelListService = mock(ArtikelListService.class);
+    private final IUserRepo testUserRepo = mock(IUserRepo.class);
+    private final UserService testUserService = new UserService(testUserRepo);
+    private final ArtikelListController testArtikelListController =
+            new ArtikelListController(mockedArtikelListService, testUserService);
+
+    @Test
+    void getAll() {
+        //Given
+        Principal testPrincipal = () -> "Susi";
+        User testUser = User.newUser("Susi", "su@si.cd", "Schulz",
+                "suesue", "su123", List.of(
+                        new SimpleGrantedAuthority("API_READWRITE")));
+
+        List<Artikel> artikels = Collections.emptyList();
+        String userId = "userId";
+        String listId = "listId";
+        ArtikelList testArtikelList = ArtikelList.newArtikelList(listId, artikels, userId);
+        List<ArtikelList> testArtikelListen = List.of(testArtikelList);
+        //When
+        when(testUserRepo.findUserByUsername("Susi")).thenReturn(testUser);
+        when(mockedArtikelListService.getAllArtikelLists(testUser.getUsername()))
+                .thenReturn(testArtikelListen);
+        //Then
+        assertEquals(testArtikelListen, testArtikelListController.getAll(testPrincipal));
+    }
+
+    @Test
+    void addArtikelList() throws CloneNotSupportedException {
+        //Given
+        Principal testPrincipal = () -> "Susi";
+        User testUser = User.newUser("Susi", "su@si.cd", "Schulz",
+                "suesue", "su123", List.of(
+                        new SimpleGrantedAuthority("API_READWRITE")));
+
+        ListDto testListDto = new ListDto("testListName");
+
+        List<Artikel> artikels = Collections.emptyList();
+        String userId = "userId";
+        String listId = "listId";
+        ArtikelList testArtikelList = ArtikelList.newArtikelList(listId, artikels, userId);
+        //When
+        when(testUserRepo.findUserByUsername("Susi")).thenReturn(testUser);
+        when(mockedArtikelListService.addArtikelList(testUser.getUsername(), testListDto.getListName()))
+                .thenReturn(testArtikelList);
+        //Then
+        assertEquals(testArtikelList, testArtikelListController.addArtikelList(testPrincipal, testListDto));
```

Comment:
I would save the result of the test call and then check the result.

```suggestion
        ArtikeList actual = testArtikelListController.addArtikelList(testPrincipal, testListDto);
        //Then
        assertEquals(testArtikelList, actual);
```